src/tasks/task-angular-momentum-equality.cpp

```diff
@@ -0,0 +1,174 @@
+
+namespace tsid
+{
+  namespace tasks
+  {
+    using namespace math;
+    using namespace trajectories;
+    using namespace pinocchio;
+
+    TaskAMEquality::TaskAMEquality(const std::string & name,
+                                     RobotWrapper & robot):
+      TaskBase(name, robot),
+      m_constraint(name, 3, robot.nv())
+    {
+      m_Kp.setZero(3);
+      m_Kd.setZero(3);
+      m_L_error.setZero(3);
+      m_dL_error.setZero(3);
+      m_L.setZero(3);
+      m_dL.setZero(3);
+      m_dL_des.setZero(3);
+      m_mask.setOnes(3);
+      mask(m_mask);
+    }
+
+    int TaskAMEquality::dim() const
+    {
+      return (int)m_mask.sum();
+    }
+    const Vector3 & TaskAMEquality::mask() const
+    {
+      return m_mask;
+    }
+
+    void TaskAMEquality::mask(const Vector3 & m)
+    {
+      //TODO: ggory15
+      m_mask = m;
+      const Vector::Index dim = static_cast<Vector::Index>(m.sum());
+      Matrix S = Matrix::Zero(dim, m_robot.nv());
+      m_activeAxes.resize(dim);
+      unsigned int j=0;
+      for(unsigned int i=0; i<m.size(); i++)
+        if(m(i)!=0.0)
+        {
+          assert(m(i)==1.0);
+          m_activeAxes(j) = i;
+          j++;
+        }
+      m_constraint.resize((unsigned int)dim, m_robot.nv());
+    }
+
+
+    const Vector3 & TaskAMEquality::Kp(){ return m_Kp; }
+
+    const Vector3 & TaskAMEquality::Kd(){ return m_Kd; }
+
+    void TaskAMEquality::Kp(ConstRefVector Kp)
+    {
+      assert(Kp.size()==3);
+      m_Kp = Kp;
+    }
+
+    void TaskAMEquality::Kd(ConstRefVector Kd)
+    {
+      assert(Kd.size()==3);
+      m_Kd = Kd;
+    }
+
+    void TaskAMEquality::setReference(const TrajectorySample & ref)
+    {
+      m_ref = ref;
+    }
+
+    const TrajectorySample & TaskAMEquality::getReference() const
+    {
+      return m_ref;
+    }
+
+    const Vector3 & TaskAMEquality::getDesireddMomentum() const
+    {
+      return m_dL_des;
+    }
+
+    Vector3 TaskAMEquality::getdMomentum(ConstRefVector dv) const
+    {
+      return m_constraint.matrix()*dv - m_drift;
+    }
+
+    const Vector3 & TaskAMEquality::momentum_error() const
+    {
+      return m_L_error;
+    }
+
+    const Vector3 & TaskAMEquality::momentum() const
+    {
+      return m_L;
+    }
+    const Vector & TaskAMEquality::momentum_ref() const
+    {
+      return m_ref.pos;
+    }
+
+    const Vector & TaskAMEquality::dmomentum_ref() const
+    {
+      return m_ref.acc;
+    }
+
+    const ConstraintBase & TaskAMEquality::getConstraint() const
+    {
+      return m_constraint;
+    }
+
+    const ConstraintBase & TaskAMEquality::compute(const double ,
+                                                    ConstRefVector q,
+                                                    ConstRefVector v,
+                                                    const Data & data)
+    {
+      // Compute errors
+      // Get momentum jacobian
+      Data data_ccrba = data;
+      pinocchio::ccrba(m_robot.model(), data_ccrba, q, v);
```

Comment:
You shouldn't call Pinocchio's function directly, but use the robot wrapper instead.